src/lib/updating-element.ts

```diff
@@ -462,7 +483,48 @@ export abstract class UpdatingElement extends HTMLElement {
    * @returns {Element|DocumentFragment} Returns a node into which to render.
    */
   protected createRenderRoot(): Element|ShadowRoot {
-    return this.attachShadow({mode : 'open'});
+    const shadowRoot = this.attachShadow({mode : 'open'});
+    this.createRenderRootStyles(shadowRoot);
+    return shadowRoot;
+  }
+
+  /**
+   * Applies styling to the element shadowRoot using the `static get styles`
+   * property. Styling will apply using `shadowRoot.adoptedStyleSheets` where
+   * available and will fallback otherwise.
+   */
+  protected createRenderRootStyles(shadowRoot: ShadowRoot) {
+    let styles = (this.constructor as typeof UpdatingElement).styles;
```

Comment:
Check if styles is undefined and return to avoid extra works on elements that don't declare styles?

Reply:
Thanks, done.